bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -95,99 +145,273 @@ impl Config {
+    pub fn wasi_enabled(&self) -> bool {
+        unsafe {
+            wasmedge::WasmEdge_ConfigureHasHostRegistration(
+                self.ctx,
+                wasmedge::WasmEdge_HostRegistration_Wasi,
+            )
+        }
+    }
+
+    /// Checks if host registration wasmedge process turns on or not.
+    pub fn wasmedge_process_enabled(&self) -> bool {
+        unsafe {
+            wasmedge::WasmEdge_ConfigureHasHostRegistration(
+                self.ctx,
+                wasmedge::WasmEdge_HostRegistration_WasmEdge_Process,
+            )
+        }
+    }
+
+    /// Enables host registration WasmEdge process.
+    pub fn enable_wasmedge_process(self) -> Self {
+        unsafe {
+            wasmedge::WasmEdge_ConfigureAddHostRegistration(
+                self.ctx,
+                wasmedge::WasmEdge_HostRegistration_WasmEdge_Process,
+            )
+        };
+        self
+    }
+
+    /// Sets the maximum number of the memory pages available.
+    pub fn set_max_memory_pages(self, count: u32) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureSetMaxMemoryPage(self.ctx, count) };
         self
     }
 
-    /// Get the page limit of memory instances.
+    /// Returns the number of the memory pages available.
     pub fn get_max_memory_pages(&self) -> u32 {
         unsafe { wasmedge::WasmEdge_ConfigureGetMaxMemoryPage(self.ctx) }
     }
 
     // For AOT compiler
 
-    /// Set the optimization level of AOT compiler.
+    /// Sets the optimization level of AOT compiler.
     pub fn set_optimization_level(self, opt_level: CompilerOptimizationLevel) -> Self {
         unsafe {
             wasmedge::WasmEdge_ConfigureCompilerSetOptimizationLevel(self.ctx, opt_level as u32)
         };
         self
     }
 
-    /// Get the optimization level of AOT compiler.
+    /// Returns the optimization level of AOT compiler.
     pub fn get_optimization_level(&self) -> CompilerOptimizationLevel {
         let level = unsafe { wasmedge::WasmEdge_ConfigureCompilerGetOptimizationLevel(self.ctx) };
         level.into()
     }
 
-    /// Set the output binary format of AOT compiler.
+    /// Sets the output binary format of AOT compiler.
     pub fn set_compiler_output_format(self, format: CompilerOutputFormat) -> Self {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerSetOutputFormat(self.ctx, format as u32) };
         self
     }
 
-    /// Get the output binary format of AOT compiler.
+    /// Returns the output binary format of AOT compiler.
     pub fn get_compiler_output_format(&self) -> CompilerOutputFormat {
         let value = unsafe { wasmedge::WasmEdge_ConfigureCompilerGetOutputFormat(self.ctx) };
         value.into()
     }
 
-    /// Set the dump IR boolean value of AOT compiler.
-    pub fn dump_ir(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetDumpIR(self.ctx, enable) };
+    /// Sets the dump IR option of AOT compiler.
+    pub fn enable_dump_ir(self, flag: bool) -> Self {
```

Comment:
The `enable_` prefix is very confusing. Looks like this is just set the `dump_ir` option to `flag`.
I prefer to use `set` instead. So do the following `enable_`.

Reply:
Accepted. Will refactor the function name. Thanks!

Reply:
How about just using the original name `dump_ir`? It seems much better than `enable_dump_ir` and `set_dump_ir`.

Reply:
> How about just using the original name `dump_ir`? It seems much better than `enable_dump_ir` and `set_dump_ir`.

Agree. Simple and clear.